src/components/Attachment/Attachment.tsx

```diff
@@ -0,0 +1,445 @@
+import React, { PropsWithChildren } from 'react';
+import ReactPlayer, { ReactPlayerProps } from 'react-player';
+
+import {
+  AttachmentActionsProps,
+  AttachmentActions as DefaultAttachmentActions,
+} from './AttachmentActions';
+import { AudioProps, Audio as DefaultAudio } from './Audio';
+import { CardProps, Card as DefaultCard } from './Card';
+import {
+  FileAttachment as DefaultFile,
+  FileAttachmentProps,
+} from './FileAttachment';
+
+import {
+  Gallery as DefaultGallery,
+  ImageComponent as DefaultImage,
+  GalleryProps,
+  ImageProps,
+} from '../Gallery';
+
+import type { Attachment as StreamAttachment } from 'stream-chat';
+
+import type { UnknownType } from '../../../types/types';
+
+export const SUPPORTED_VIDEO_FORMATS = [
+  'video/mp4',
+  'video/ogg',
+  'video/webm',
+  'video/quicktime',
+];
+
+export type AttachmentProps = {
+  /**
+   * The attachment to render
+   * See [Attachment structure](https://getstream.io/chat/docs/#message_format)
+   **/
+  attachments: ExtendedAttachment[];
+  /**
+   * Handle for click on action on Attachment
+   */
+  actionHandler?: (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
```

Comment:
should this type be the same as the one below? also the below comment is better so can use that one

Reply:
switched the comment and those actionhandlers should definitely be the same type. Updated both to React.MouseEvent<HTMLButtonElement, MouseEvent>.